packages/@aws-cdk/aws-signer/lib/signing-profile.ts

```diff
@@ -0,0 +1,178 @@
+import { Duration, IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnSigningProfile } from './signer.generated';
+
+/**
+ * Platforms that are allowed with signing config.
+ * @see https://docs.aws.amazon.com/signer/latest/developerguide/gs-platform.html
+ */
+export class Platform {
+  /**
+   * Specification of signature format and signing algorithms for AWS IoT Device.
+   */
+  public static readonly AWS_IOT_DEVICE_MANAGEMENT_SHA256_ECDSA = new Platform('AWSIoTDeviceManagement-SHA256-ECDSA');
+
+  /**
+   * Specification of signature format and signing algorithms for AWS Lambda.
+   */
+  public static readonly AWS_LAMBDA_SHA384_ECDSA = new Platform('AWSLambda-SHA384-ECDSA');
+
+  /**
+   * Specification of signature format and signing algorithms with
+   * SHA1 hash and RSA encryption for Amazon FreeRTOS.
+   */
+  public static readonly AMAZON_FREE_RTOS_TI_CC3220SF = new Platform('AmazonFreeRTOS-TI-CC3220SF');
+
+  /**
+   * Specification of signature format and signing algorithms with
+   * SHA256 hash and ECDSA encryption for Amazon FreeRTOS.
+   */
+  public static readonly AMAZON_FREE_RTOS_DEFAULT = new Platform('AmazonFreeRTOS-Default');
+
+  /**
+   * The id of signing platform.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-signer-signingprofile.html#cfn-signer-signingprofile-platformid
+   */
+  public readonly platformId: string;
+
+  private constructor(platformId: string) {
+    this.platformId = platformId;
+  }
+}
+
+/**
+ * A Signer Profile
+ */
+export interface ISigningProfile extends IResource {
+  /**
+   * The ARN of the signing profile.
+   * @attribute
+   */
+  readonly signingProfileArn: string;
+
+  /**
+   * The name of signing profile.
+   * @attribute
+   */
+  readonly signingProfileProfileName: string;
+
+  /**
+   * The version of signing profile.
+   * @attribute
+   */
+  readonly signingProfileProfileVersion: string;
+
+  /**
+   * The ARN of signing profile version.
+   * @attribute
+   */
+  readonly signingProfileProfileVersionArn: string;
```

Comment:
Let's make these names nice to read.

```suggestion
   * @attribute signingProfileProfileName
   */
  readonly signingProfileName: string;

  /**
   * The version of signing profile.
   * @attribute signingProfileProfileVersion
   */
  readonly signingProfileVersion: string;

  /**
   * The ARN of signing profile version.
   * @attribute signingProfileProfileVersionArn
   */
  readonly signingProfileVersionArn: string;
```